### ISSUE TITLE
I would like to scale my frontview to half when it moves to the side....

### ISSUE THREAD

**Issue by gabrielepalma** [John-Lluch/SWRevealViewController#395]
So, i'm using SWRevealViewController and all is fine and beautiful. Now i would like to create an effect where, when a front view moves to the side to leave space to the back view it also resizes to half it's original size and center itself vertically.

So far i tried a couple different ways. I tried applying a transform inside willMoveToPosition delegate (depending on the new FrontViewPosition a scale 0.5f or an identity transform) but the frame SWReveal was using for the view was sticky so it left a void on the whole height. I also tried modifying the frame directly inside SWReveal sources like that:

CGRect frame;
if(position == FrontViewPositionRight) frame = CGRectMake(xLocation, bounds.size.height/4, bounds.size.width/2, bounds.size.height/2);
else frame = CGRectMake(xLocation, 0.0f, bounds.size.width, bounds.size.height);

and that made the view resize, of course, but not scale so the layout was all messed up. Also, transforming frontViewController.view makes a painful glitch where the entire view disappear leaving a black void, and i'm guessing it is because that view is a navigationController, while if i try scaling the view inside the viewController it almost work but is glitchy (sometimes end up in the wrong position after scaling back with identity, plus the navigationController is not scaled so layout is still not right).

Being unable to scale the navigationController was probably the most annoying part, as i suppose i could have played with frames in SWReveal to eventually make it place the collapsed frontViewController appropriately.

Anybody attempted anything similar and has suggestions to give?

**Comment by rjuzaszek** [comment]
have You figure out anything maybe?

**Comment by iDevelopper** [comment]
Do you want to do an animation like LGMenuViewController?

**Comment by rjuzaszek** [comment]
exactly, thanks!

**Comment by iDevelopper** [comment]
I don't think it will be possible with SWRevealViewController unless you do a fork. Why do not use LGMenuViewController?

**Comment by rjuzaszek** [comment]
I do now, I was close to similar effect but still needed some fixes hard to resolve, LGMenuViewController works pretty well, and apparently has much more options than SWReveal
